theano/sandbox/gpuarray/basic_ops.py

```diff
@@ -312,254 +309,29 @@ def infer_shape(self, node, xshp):
 
     def c_code(self, node, name, inputs, outputs, sub):
         return """
+        PyGpuArrayObject *%(name)s_tmp;
+        %(name)s_tmp = PyArray_GETCONTIGUOUS(%(inp)s);
```

Comment:
Because pygpu_fromhostdata() does not have access to the ndarray object and currently tries to copy the data as-is.  I've realized that it's not a good idea and made a ticket to change that in https://github.com/Theano/libgpuarray/issues/90

I made a ticket that might help at